kubespawner/spawner.py

```diff
@@ -34,6 +35,7 @@
 from jupyterhub.utils import exponential_backoff
 from jupyterhub.traitlets import Command
 from kubernetes.client.rest import ApiException
+from urllib3.exceptions import ReadTimeoutError
```

Comment:
(Consider my comments a project-learning experience!) 

Should you specify urllib3 in the setup.py as well?  Of course it's going to come in as a transitive dep of 'kubernetes' - I don't know whether the policy would be to explicitly require it.

Reply:
It appears that ReadTimeoutError is used in reflector.py still, but not in spawner.py. I personally think it should be an explicit dependency for kubespawner if we actually have it explicitly imported still.

> This also means we don't have to import urllib3 anymore, so one less implicit dependency!

```suggestion
```

Reply:
I opened https://github.com/jupyterhub/kubespawner/pull/437 for that